### ISSUE TITLE
Playlist: error in Safari

### ISSUE THREAD

**Issue by sprungmarker** [mediaelement/mediaelement-plugins#129]
Playlist error in Safari 11.1 - is not generating playlist and gets error - line 311:

` undefined is not an object (evaluating 'childNode.tagName.toLowerCase')`

This error is also in the demo version.

**Comment by sprungmarker** [comment]
Is there any progress for safari? You cannot use playlist Plugin for Safari ...

**Comment by berryamin** [comment]
Looks the same as https://github.com/mediaelement/mediaelement/issues/2251

**Comment by thatcher** [comment]
still broken for me.  any news?  the real problem to me is this:

```
var children = t.mediaFiles || t.media.originalNode.children;

for (var i = 0, total = children.length; i < total; i++) {
    var childNode = children[i];

    if (childNode.tagName.toLowerCase() === 'source') {
```

Note that `t.mediaFiles` is a simple Array of Objects so iterating over it and calling tagName will _always_ fail.  

Do you agree @rafa8626 ?

**Comment by thatcher** [comment]
I was able to fix this by changing it to:

```
createPlayList_: function createPlayList_() {
    var t = this;

    t.playlist = t.options.playlist.length ? t.options.playlist : [];

    if (!t.playlist.length) {
        if (t.mediaFiles && t.mediaFiles.length){
            t.playlist = t.mediaFiles;
        } else {
            var children = t.media.originalNode.children;

```

**Comment by berryamin** [comment]
Thanks for PR. Will be included in release 2.5.1 to come out soon.
